src/downloader.cpp

```diff
@@ -49,10 +49,11 @@ namespace powerloader
         if (!dl_target)
             return;
 
-        auto mirrors = ctx.mirror_map.get_mirrors(dl_target->base_url());
-        if (!mirrors.empty())
+        auto mirrors = ctx.mirror_map.get_mirrors(dl_target->mirror_name());
+        if (mirrors.empty())
         {
-            dl_target->clear_base_url();
+            // TODO proper error handling here! :)
```

Comment:
What error handling is missing?

Reply:
Not throwing runtime error but something more proper.